flood_adapt/integrator/sfincs_adapter.py

```diff
@@ -88,13 +347,280 @@ def add_wind_forcing(
+                UnitTypesLength("meters")
+            )
+
+        # par1 is the overflow coefficient for weirs
+        gdf_floodwall["par1"] = 0.6
+
+        # HydroMT function: create floodwall
+        self._model.setup_structures(structures=gdf_floodwall, stype="weir", merge=True)
+
+    def _add_measure_greeninfra(self, green_infrastructure: GreenInfrastructure):
+        # HydroMT function: get geodataframe from filename
+        if green_infrastructure.attrs.selection_type == "polygon":
+            polygon_file = Database().input_path.joinpath(
+                green_infrastructure.attrs.polygon_file
+            )
+        elif green_infrastructure.attrs.selection_type == "aggregation_area":
+            # TODO this logic already exists in the Database controller but cannot be used due to cyclic imports
+            # Loop through available aggregation area types
+            for aggr_dict in self._site.attrs.fiat.aggregation:
+                # check which one is used in measure
+                if (
+                    not aggr_dict.name
+                    == green_infrastructure.attrs.aggregation_area_type
+                ):
+                    continue
+                # load geodataframe
+                aggr_areas = gpd.read_file(
+                    Database().static_path / "site" / aggr_dict.file,
+                    engine="pyogrio",
+                ).to_crs(4326)
+                # keep only aggregation area chosen
+                polygon_file = aggr_areas.loc[
+                    aggr_areas[aggr_dict.field_name]
+                    == green_infrastructure.attrs.aggregation_area_name,
+                    ["geometry"],
+                ].reset_index(drop=True)
+        else:
+            raise ValueError(
+                f"The selection type: {green_infrastructure.attrs.selection_type} is not valid"
+            )
+
+        gdf_green_infra = self._model.data_catalog.get_geodataframe(
+            polygon_file,
+            geom=self._model.region,
+            crs=self._model.crs,
+        )
+
+        # Make sure no multipolygons are there
+        gdf_green_infra = gdf_green_infra.explode()
+
+        # Volume is always already calculated and is converted to m3 for SFINCS
+        height = None
+        volume = green_infrastructure.attrs.volume.convert(UnitTypesVolume("m3"))
+
+        # HydroMT function: create storage volume
+        self._model.setup_storage_volume(
+            storage_locs=gdf_green_infra, volume=volume, height=height, merge=True
+        )
+
+    def _add_measure_pump(self, pump: Pump):
+        """Add pump to sfincs model.
+
+        Parameters
+        ----------
+        pump : PumpModel
+            pump information
+        """
+        polygon_file = Database().input_path.joinpath(pump.attrs.polygon_file)
+        # HydroMT function: get geodataframe from filename
+        gdf_pump = self._model.data_catalog.get_geodataframe(
```

Comment:
ideally you could get pump discharges also from the shape file like the heights for the floodwall. Nice to have for later

Reply:
The discharge is available in this function as pump.attrs.discharge! putting it in the shapefile shouldnt be a problem though